packages/slate-plugins/src/elements/code-block/CodeBlockPlugin.ts

```diff
@@ -2,19 +2,28 @@
  * Enables support for pre-formatted code blocks.
  */
 import { SlatePlugin } from '@udecode/slate-plugins-core';
+import { setDefaults } from '../../common/utils/setDefaults';
 import { decorateCodeBlock } from './decorateCodeBlock';
+import { DEFAULTS_CODE_BLOCK } from './defaults';
 import { deserializeCodeBlock } from './deserializeCodeBlock';
 import { onKeyDownCodeBlock } from './onKeyDownCodeBlock';
 import { renderElementCodeBlock } from './renderElementCodeBlock';
 import { renderLeafCodeBlock } from './renderLeafCodeBlock';
-import { CodeBlockPluginOptions } from './types';
+import {
+  CodeBlockContainerPluginOptions,
+  CodeBlockPluginOptions,
+} from './types';
 
 export const CodeBlockPlugin = (
-  options?: CodeBlockPluginOptions
-): SlatePlugin => ({
-  renderElement: renderElementCodeBlock(options),
-  renderLeaf: renderLeafCodeBlock(),
-  deserialize: deserializeCodeBlock(options),
-  decorate: decorateCodeBlock(),
-  onKeyDown: onKeyDownCodeBlock(options),
-});
+  options?: CodeBlockPluginOptions & CodeBlockContainerPluginOptions
+): SlatePlugin => {
+  const { code_block_container } = setDefaults(options, DEFAULTS_CODE_BLOCK);
+  return {
+    renderElement: renderElementCodeBlock(options),
+    renderLeaf: renderLeafCodeBlock(),
+    deserialize: deserializeCodeBlock(options),
+    decorate: decorateCodeBlock(),
+    onKeyDown: onKeyDownCodeBlock(options),
+    voidTypes: [code_block_container.type],
```

Comment:
I mostly followed this, but to remain consistent with the naming conventions for everything else, I instead went with CodeBlockContainerPlugin and CodeBlockPlugin. My thinking was that this would make it so people could continue using CodeBlockPlugin as is, or choose the container version by opting-in. I'm happy to change it to your suggestion, but figured I'd see what you think first.